runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingStepMetricsContainer.java

```diff
@@ -68,6 +74,14 @@ public class StreamingStepMetricsContainer implements MetricsContainer {
   private MetricsMap<KV<MetricName, HistogramData.BucketType>, HistogramCell> perWorkerHistograms =
       new MetricsMap<>(HistogramCell::new);
 
+  private Map<MetricName, Instant> perWorkerCountersByFirstStaleTime = new ConcurrentHashMap<>();
```

Comment:
make final unless it can be reassigned
`private final Map<MetricName, Instant> perWorkerCountersByFirstStaleTime = new ConcurrentHashMap<>();`

Reply:
made it `final` but I am instantiating the variable in the constructor so I can inject the value for testing.